src/api/model/modelApi.ts

```diff
@@ -14,57 +14,71 @@
  * under the License.
  */
 
-import { TransactionApi } from '../transaction/transactionApi';
-import {
-  InstallAction,
-  ListSourceAction,
-  Model,
-  ModifyWorkspaceAction,
-} from '../transaction/types';
+import { ExecAsyncApi } from '../query/execAsyncApi';
+import { Model } from '../transaction/types';
+export class ModelApi extends ExecAsyncApi {
+  async installModels(
+    database: string,
+    engine: string,
+    models: Model[],
+    async = false,
+  ) {
+    const queries = models.map(model => {
+      return `def insert:rel:catalog:model["${model.name}"] = """${model.value}"""`;
+    });
 
-export class ModelApi extends TransactionApi {
-  async installModels(database: string, engine: string, models: Model[]) {
-    const action: InstallAction = {
-      type: 'InstallAction',
-      sources: models,
-    };
+    const rsp = async
+      ? await this.execAsync(database, engine, queries.join('\n'), [], false)
+      : await this.exec(database, engine, queries.join('\n'), [], false);
 
-    return await this.runActions(database, engine, [action], false);
+    return rsp;
   }
 
   async listModels(database: string, engine: string) {
```

Comment:
This returns an array of models. Which's convenient ofc, but it swallows the diagnostics/problems. You'd have to issue a separate query dummy query to get them.

Not sure what to do here. I guess ideally we want to parse diagnostics into an object. So that we can return something like `{ models: [{...}], diagnostics: [{...}]}`. But it's too much for this PR obviously.

Reply:
is there an easy way to get diagnostics other than `rsp.problems` field ?
I added a model test with value `def foo =` and tried to get diagnostics as below
```
def output = rel:catalog:diagnostic
```
which returns nothing

Reply:
You can send any query with errors and you'll get them in the results.

```
def output = 123;
```
will do